### ISSUE TITLE
`logits_to_keep` is not working in training mode

### ISSUE THREAD

**Issue by Tcc0403** [linkedin/Liger-Kernel#694]
### 🐛 Describe the bug

`logits_to_keep` should work in training mode as HF transformers [does](https://github.com/huggingface/transformers/blob/v4.51.3/src/transformers/models/llama/modeling_llama.py#L836).

Most models in Liger only trim logits in inference mode:
https://github.com/linkedin/Liger-Kernel/blob/05b43a14913ced3776aa3fc50020089b8c0d63c1/src/liger_kernel/transformers/model/gemma.py#L217-L219

We should trim hidden states instead to ensure that Liger and HF have same behaviors when `logits_to_keep` is given. See how Liger's gemma3 does:
https://github.com/linkedin/Liger-Kernel/blob/05b43a14913ced3776aa3fc50020089b8c0d63c1/src/liger_kernel/transformers/model/gemma3.py#L106-L110

### Versions

liger_kernel==0.5.8
